decode.go

```diff
@@ -215,6 +214,10 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 	// Signed integers may be encoded either zigzag-varint or fixed
 	// Note that protobufs don't support 8- or 16-bit ints.
 	case reflect.Int, reflect.Int32, reflect.Int64:
+		if val.Kind() == reflect.Int && val.Type().Size() < 8 {
+			panic("Detected a 32bit machine, please either use int64 or int32. " +
+				"However, ensure you are using the same size being used on the sender side not to lose information.")
```

Comment:
`However...` is not necessary. If somebody uses `protobuf`, we can suppose he's using it on both the sender and receiver side. So if he changes a field to `int32`, then both the sender and the receiver will get it correct.

Reply:
I think panic is too strong, decode takes user input so an attacker may create a bad input, send it to the decoder, and cause it to crash

Reply:
Right - I didn't think about that attack scenario. I just wanted to make sure it doesn't pass. So `return errors.New` makes more sense here.